source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java

```diff
@@ -3,135 +3,116 @@
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.graphics.Texture;
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
-import com.badlogic.gdx.scenes.scene2d.Actor;
-import com.badlogic.gdx.scenes.scene2d.InputEvent;
-import com.badlogic.gdx.scenes.scene2d.InputListener;
 import com.badlogic.gdx.scenes.scene2d.ui.*;
-import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
-import com.badlogic.gdx.scenes.scene2d.utils.Drawable;
+
 import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
 import com.csse3200.game.inventory.Inventory;
 import com.csse3200.game.inventory.items.AbstractItem;
 import com.csse3200.game.services.ServiceLocator;
 import com.csse3200.game.ui.DialogueBox;
 import com.csse3200.game.ui.UIComponent;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
- * PlayerInventoryHotbarDisplay is a UI component that displays players quick selection panel for
- * better user interaction during gameplay
- * It creates a table of first n inventory slots that can display items, handle item usage,
- * and update dynamically when the inventory changes.
+ * UI component for displaying the player's quick-selection hotbar.
+ * It manages a table with a fixed number of inventory slots that display items,
+ * handle item usage, and update dynamically based on inventory changes.
  */
-
 public class PlayerInventoryHotbarDisplay extends UIComponent {
 
-    private static final Logger logger = LoggerFactory.getLogger(PlayerInventoryHotbarDisplay.class);
-    private final Skin skinSlots=new Skin(Gdx.files.internal("Inventory/skinforslot.json"));
-    private final Table table;
+
+    private final Skin skinSlots = new Skin(Gdx.files.internal("Inventory/skinforslot.json")); //created by @PratulW5
+    private final Table table = new Table();
     private final Inventory inventory;
     private final ImageButton[] hotBarSlots;
-    private final int numberOfSlots;
-    private final Texture hotBarTexture;
-    private boolean toggleHotbar = true;
-    PlayerInventoryDisplay playerInventoryDisplay;
+
+    private final int hotBarCapacity;
+    private final Texture hotBarTexture = new Texture("Inventory/hotbar.png");//created by @PratulW5
+
+    private final PlayerInventoryDisplay playerInventoryDisplay;
+
+    /**
+     * Constructs a PlayerInventoryHotbarDisplay with the specified hotbar capacity and inventory.
+     *
+     * @param hotBarCapacity Number of slots in the hotbar
+     * @param inventory      Player's inventory
+     * @param playerInventoryDisplay Inventory display manager
+     */
     public PlayerInventoryHotbarDisplay(int hotBarCapacity, Inventory inventory, PlayerInventoryDisplay playerInventoryDisplay) {
-        if (hotBarCapacity < 1) {
-            throw new IllegalArgumentException("Inventory Hotbar dimensions must be more than one!");
-        }
+        if (hotBarCapacity < 1) throw new IllegalArgumentException("Inventory Hotbar dimensions must be more than one!");
```

Comment:
just call this capacity (also throw in a check to ensure capacity <= inventory.getCapacity())